src/psi-dynamics.scm

```diff
@@ -0,0 +1,279 @@
+(use-modules (opencog nlp) (opencog nlp chatbot))
+
+(load "express.scm") ; For random pos and neg expressions
+(load "faces.scm")
+(load "self-model.scm") ; For soma-state def
+(load "orchestrate.scm") ; For DefinedPredicate "Show expression"
+
+; Param setting
+(define valence-activation-level .5)
+
+(define psi-verbose #t)
+(define no-blender #f)
+
+(if no-blender
+    (python-eval "execfile('/usr/local/share/opencog/python/atomic-dbg.py')"))
+
+
+(define prev-value-node (Concept "previous value"))
+(define current-sentence-node (Concept "current sentence"))
+
+; Temporary call needed to load dynamics code while it's in dev phase
+(load-openpsi-in-development)
+
+; Function called by OpenPsi when it believes expression command updates should
+; occur based on event detection and subsequent OpenPsi variable updating
+(define (psi-expression-callback)
+	(define arousal (psi-get-arousal))
+	(define pos-valence (psi-get-pos-valence))
+	(define neg-valence (psi-get-neg-valence))
+	;(if psi-verbose (display "psi-dynamics expression callback called\n"))
+
+	; For now we are doing something simple - do a random positive or negative
+	; expression based on valence and arousal.
+	; Later arousal will be used to modulate intensity of the expression
+	; Todo: How to handle when both pos and neg valence are high ? Expressing
+	; both for now, which may or may not be a good thing, but probably
+	; interesting nonetheless.
+	(if (>= pos-valence valence-activation-level)
+	    (begin
+            ;( (do-catch do-random-positive-expression))
+            (if psi-verbose
+                (display "psi-dynamics: doing positive expression\n"))
+            (if (not no-blender)
+                (be-happy pos-valence)
+                ;(do-random-positive-expression)
+            )
+        )
+	)
+	(if (>= neg-valence valence-activation-level)
+	    (begin
+            ;( (do-catch do-random-negative-expression))
+            (if psi-verbose
+                (display "psi-dynamics: doing negative expression\n"))
+            (if (not no-blender)
+                (be-sad neg-valence)
+                ;(do-random-negative-expression)
+		    )
+	    )
+	)
+
+	; Update psi-emotion-states
+	; Keeping it simple for now
+	; For happy perhaps pos-valence * arousal or weighted average?
+	; Todo: Map out other emotions
+	(psi-set-value! psi-happy (* pos-valence arousal))
+)
+
+; Register the expression callback function with OpenPsi
+(psi-set-expression-callback! psi-expression-callback)
+
+; ------------------------------------------------------------------
+; Functions to initiate random positive and negative epxerssions
+
+(define (do-random-positive-expression intensity)
+    (define expression
+       (cog-execute!
+          (PutLink (DefinedSchemaNode "Pick random expression")
+             (ConceptNode "positive"))))
+     (cog-evaluate! (Put (DefinedPredicate "Show expression")
+         (ListLink expression (Number 8) (Number intensity)))))
```

Comment:
well, there already exist macros to do this stuff, and they also pick random lengths of time over which to run these expressions.   In fact, they are all set up in the configuration file, so that you would not have to write any new code, either here, or below, to add this stuff.

If you really want a hard-coded value of 8 seconds for the expression duriation, just set both min and max to 8 in the config file.

So .. for general code-maintenance purposes, I think it would be better to just delete this and the 3 other functions below, and just set it all up with the config file.